tests/testlibrary.c

```diff
@@ -1542,6 +1542,58 @@ test_list_updates (void)
   g_assert_true (res);
 }
 
+static void
+test_list_updates_offline (void)
+{
+  if (g_test_subprocess ())
+    {
+      g_autoptr(FlatpakInstallation) inst = NULL;
+      g_autoptr(GError) error = NULL;
+      g_autoptr(GPtrArray) refs = NULL;
+      g_autoptr(FlatpakInstalledRef) runtime_ref = NULL;
+      gboolean res;
+
+      inst = flatpak_installation_new_user (NULL, &error);
+      g_assert_no_error (error);
+
+      /* Install a runtime */
+      runtime_ref = flatpak_installation_install (inst,
+                                                  repo_name,
+                                                  FLATPAK_REF_KIND_RUNTIME,
+                                                  "org.test.Platform",
+                                                  NULL, NULL, NULL, NULL, NULL,
+                                                  &error);
+      g_assert_no_error (error);
+      g_assert (FLATPAK_IS_INSTALLED_REF (runtime_ref));
+
+      /* Cut off the subprocess from the network the same way bubblewrap does it */
+      unshare (CLONE_NEWNET);
```

Comment:
This doesn't work. You must have CAP_SYS_ADMIN capabilities to be able to unshare something. Bubblewrap can do so, because it is in a new user namespace which is has full capabilities within (or because it is setuid root).

Reply:
It seems to work though. The CI failure happens after this test passes:
```
PASS: testlibrary 20 /library/list-updates-offline
PASS: testlibrary 21 /library/transaction
PASS: testlibrary 22 /library/transaction-install-uninstall
ERROR: testlibrary - Bail out! flatpak:ERROR:tests/testlibrary.c:2630:add_new_remote2: assertion failed (suggested_name == "my-little-repo"): ("test-repo-1" == "my-little-repo")
```

Reply:
Oh, I guess because the subsequent tests are run with networking still disabled.

Reply:
`setns(2)` seems to imply it's the re-assocation of a process with a namespace that requires CAP_SYS_ADMIN, so maybe we could have two tests, testlibrary and testlibrary-offline.

Reply:
It turned out that assertion failure is not because the rest of the tests are run without networking, but because the subprocess is run with the same cwd as the parent process, and the subprocess overwrites a flatpakrepo with a URL with a different port, since `global_setup()` happens again.

Reply:
So I pushed a change to change the cwd in the subprocess.

Reply:
Still, unshare() doesn't work like that, and in fact the test passes even if I comment out the unshare call like so:
```
-      unshare (CLONE_NEWNET);
+      //unshare (CLONE_NEWNET);
```
So, we're testing something else it seems.

Reply:
Right, because there's no update available for the test runtime. So we don't see one regardless of whether we're offline or not. I figured it wasn't worth creating an update since the subprocess won't see it anyway, but perhaps it's worthwhile to make sure we're testing offline behavior.

If you think we should just mangle the remote's URL instead of doing this, I'm fine with that too. I don't think libostree or flatpak handle a 404 error differently than the error you'd get when offline.

Reply:
Yeah, i think just mangling the url is fine